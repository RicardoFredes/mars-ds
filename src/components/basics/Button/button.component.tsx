import type { LinkProps } from "../Link";
import type { ButtonProps } from "./button.types";
import type { MouseEvent } from "react";

import classNames from "classnames";
import { useState } from "react";

import Icon from "@/components/basics/Icon";

import Link from "../Link";
import { ButtonSizes, ButtonTypes, ButtonVariants } from "./button.types";

const Button = ({
  children,
  className,
  iconName,
  label,
  onClick,
  type = ButtonTypes.Button,
  variant = ButtonVariants.Primary,
  size = ButtonSizes.Medium,
  as,
  isRtl = false,
  ...props
}: ButtonProps) => {
  const [isPressed, setIsPressed] = useState(false);
  const Component = as || props.href ? Link : "button";

  const handleClick = (event: MouseEvent<HTMLButtonElement & LinkProps>) => {
    if (props.disabled) return;
    setIsPressed(true);
    onClick?.(event);
  };

  const cn = classNames(
    "btn",
    `btn--${variant}`,
    `btn--size-${size}`,
    { "btn--is-disabled": props.disabled, "btn--is-pressed": isPressed, "btn--is-rtl": isRtl },
    className
  );

  const text = label || children;

  return (
    <Component
      className={cn}
      onClick={handleClick}
      type={type}
      onAnimationEnd={() => setIsPressed(false)}
      {...props}
    >
      {iconName && <Icon className="btn__icon" name={iconName} size={size} />}
      {text && <span className="btn__content">{text}</span>}
    </Component>
  );
};

export default Button;
