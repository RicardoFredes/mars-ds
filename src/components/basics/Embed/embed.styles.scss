.embed {
  width: 100%;
  display: block;
  text-align: center;
  justify-content: center;
  justify-items: center;

  .text{
    margin-bottom: 15px;
  }

  &--default {
    height: 200px;
  }

  &--landscape {
    aspect-ratio: 29.7/21;
  }

  &--portrait {
    aspect-ratio: 21/29.7;
  }
}
