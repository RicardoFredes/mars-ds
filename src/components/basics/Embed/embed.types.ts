import type { ButtonProps } from "../Button";

export const EmbedSizes = {
  Default: "default",
  Landscape: "landscape",
  Portrait: "portrait",
} as const;

interface EmbedProps extends Omit<React.HTMLProps<HTMLObjectElement>, "size"> {
  src?: string;
  type?: string;
  size?: typeof EmbedSizes[keyof typeof EmbedSizes];
  variantButton?: ButtonProps["variant"];
  target?: ButtonProps["target"];
  textButton?: string;
  messageDownload?: string;
}

export type { EmbedProps };
