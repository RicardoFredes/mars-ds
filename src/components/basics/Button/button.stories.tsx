import type { ComponentMeta, ComponentStory } from "@storybook/react";

import icons from "@/components/basics/Icon";

import Button from "./button.component";
import { ButtonSizes, ButtonVariants } from "./button.types";

export default {
  title: "Components/Button",
  component: Button,
  argTypes: {
    variant: {
      options: ButtonVariants,
      control: { type: "select" },
    },
    size: {
      options: ButtonSizes,
      control: { type: "select" },
    },
    name: {
      options: Object.keys(icons),
      control: { type: "select" },
    },
  },
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

const TemplateList: ComponentStory<typeof Button> = (args) => {
  const variants = Object.values(ButtonVariants);
  return (
    <div>
      {variants.map((variant, key) => (
        <Button key={key} {...args} variant={variant} style={{ marginBottom: 16, marginRight: 16 }}>
          {variant} button
        </Button>
      ))}
    </div>
  );
};

const DefaultArgs = {
  variant: ButtonVariants.Primary,
  size: ButtonSizes.Medium,
  children: "Quero ser aprovado!",
  disabled: false,
};

export const Default = Template.bind({});
Default.args = DefaultArgs;

export const MediumSize = TemplateList.bind({});
MediumSize.args = {
  ...DefaultArgs,
};

export const SmallSize = TemplateList.bind({});
SmallSize.args = {
  ...DefaultArgs,
  size: ButtonSizes.Small,
};

export const Disabled = TemplateList.bind({});
Disabled.args = {
  ...DefaultArgs,
  disabled: true,
};

export const Icon = TemplateList.bind({});
Icon.args = {
  ...DefaultArgs,
  iconName: "me-salva-mini",
};

export const IconSmallSize = TemplateList.bind({});
IconSmallSize.args = {
  ...DefaultArgs,
  iconName: "me-salva-mini",
  size: ButtonSizes.Small,
};
