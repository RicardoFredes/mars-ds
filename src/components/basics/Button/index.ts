/**
 * WARNING!
 * Do not change this file,
 * because it is automatically generated
 * run "yarn reindex" to update
 */

import Button from "./button.component";
export default Button;
export * from "./button.types";
