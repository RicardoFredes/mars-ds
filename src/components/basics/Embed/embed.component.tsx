import type { EmbedProps } from "./embed.types";

import classNames from "classnames";

import Text from "@/components/typographies/Text";

import Button, { ButtonSizes } from "../Button";
import { EmbedSizes } from "./embed.types";

const Embed = ({
  src,
  type = "application/pdf",
  variantButton = "primary",
  size = EmbedSizes.Default,
  target = "_blank",
  textButton = "Baixar arquivo",
  messageDownload = "Se você não consegue visualizar este documento, clique no botão abaixo e faça download!",
  className,
  ...props
}: EmbedProps) => {
  const cn = classNames("embed", `embed--${size}`, className);

  if (!src) return null;
  return (
    <object data-testid="embed" data={src} type={type} className={cn} {...props}>
      <Text data-testid="text">{messageDownload}</Text>
      <Button
        variant={variantButton}
        data-testid="button"
        href={src}
        target={target}
        size={ButtonSizes.Small}
      >
        {textButton}
      </Button>
    </object>
  );
};

export default Embed;
