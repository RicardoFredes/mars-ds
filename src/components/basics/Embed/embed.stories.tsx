import type { ComponentMeta, ComponentStory } from "@storybook/react";

import { ButtonVariants } from "../Button";
import Embed from "./embed.component";
import { EmbedSizes } from "./embed.types";

export default {
  title: "Components/Embed",
  component: Embed,
  argTypes: {
    src: {
      control: {
        type: "text",
      },
      description: "Embedded media link",
    },
    type: {
      control: {
        type: "text",
      },
      defaultValue: "application/pdf",
      description:
        "Media Types of the media embedded in the page. For examples, go <a href='https://www.iana.org/assignments/media-types/media-types.xhtml' target='_blank'>Media Types List from IANA</a>",
    },
    variantButton: {
      control: {
        type: "select",
      },
      defaultValue: ButtonVariants.Primary,
      description: "Button appearance variation",
      options: Object.values(ButtonVariants),
    },
    target: {
      control: {
        type: "text",
      },
      description: "The target attribute specifies where to open the linked document",
      defaultValue: "_target",
    },
    textButton: {
      control: {
        type: "text",
      },
      description: "Text inside the button",
      defaultValue: "Baixar arquivo",
    },
    size: {
      control: {
        type: "select",
      },
      description:
        "Defines the size and orientation of the box which is used to represent a element",
      options: Object.values(EmbedSizes),
      defaultValue: EmbedSizes.Default,
    },
    messageDownload: {
      control: {
        type: "text",
      },
      defaultValue:
        "Se você não consegue visualizar este documento, clique no botão abaixo e faça download!",
      description: "Message that appears in the mobile layout",
    },
  },
} as ComponentMeta<typeof Embed>;

const Template: ComponentStory<typeof Embed> = (args) => <Embed {...args} />;

const DefaultArgs = {
  src: "https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-v14.pdf",
};

export const Default = Template.bind({});
Default.args = DefaultArgs;

export const PdfError = Template.bind({});
PdfError.args = {
  ...DefaultArgs,
  src: "https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-.pdf",
};
