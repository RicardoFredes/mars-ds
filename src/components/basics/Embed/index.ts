/**
 * WARNING!
 * Do not change this file,
 * because it is automatically generated
 * run "yarn reindex" to update
 */

import Embed from "./embed.component";
export default Embed;
export * from "./embed.types";
