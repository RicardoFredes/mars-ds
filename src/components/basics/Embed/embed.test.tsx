import type { EmbedProps } from "./embed.types";

import { render } from "@testing-library/react";

import { ButtonVariants } from "../Button";
import Embed from "./embed.component";
import { EmbedSizes } from "./embed.types";

const makeSut = (props?: EmbedProps) => render(<Embed {...props} />);

describe("<Embed>", () => {
  it("should render component", () => {
    const wrapper = makeSut();
    expect(wrapper).toBeTruthy();
  });

  it("should render component with basic props", () => {
    const wrapper = makeSut({
      src: "https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-v14.pdf",
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("should render component with advanced props", () => {
    const wrapper = makeSut({
      src: "https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-v14.pdf",
      textButton: "Download",
      size: EmbedSizes.Landscape,
      variantButton: ButtonVariants.Naked,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should put specific className in button when "variantButton" is selected', () => {
    const wrapper = makeSut({
      src: "https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-v14.pdf",
      variantButton: ButtonVariants.Text,
    });
    const element = wrapper.getByTestId("button");
    expect(element).toHaveAttribute("class", "link btn btn--text btn--size-sm link--primary");
  });

  it('should put specific text in button when property "textButton" is not default', () => {
    const wrapper = makeSut({
      src: "https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-v14.pdf",
      textButton: "Download",
    });
    const element = wrapper.getByTestId("button");
    expect(element).toHaveTextContent("Download");
  });

  it('should put specific text in "messageDownload" property', () => {
    const wrapper = makeSut({
      src: "https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-v14.pdf",
      messageDownload: "Faça o download",
    });
    const element = wrapper.getByTestId("text");
    expect(element).toHaveTextContent("Faça o download");
  });
});
