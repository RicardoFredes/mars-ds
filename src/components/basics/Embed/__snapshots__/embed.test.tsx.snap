// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Embed> should render component with advanced props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <object
        class="embed embed--landscape"
        data="https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-v14.pdf"
        data-testid="embed"
        type="application/pdf"
      >
        <p
          class="text text--size-md"
          data-testid="text"
        >
          Se você não consegue visualizar este documento, clique no botão abaixo e faça download!
        </p>
        <a
          class="link btn btn--naked btn--size-sm link--primary"
          data-testid="button"
          href="https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-v14.pdf"
          target="_blank"
          type="button"
        >
           
          <span
            class="btn__content"
          >
            Download
          </span>
        </a>
      </object>
    </div>
  </body>,
  "container": <div>
    <object
      class="embed embed--landscape"
      data="https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-v14.pdf"
      data-testid="embed"
      type="application/pdf"
    >
      <p
        class="text text--size-md"
        data-testid="text"
      >
        Se você não consegue visualizar este documento, clique no botão abaixo e faça download!
      </p>
      <a
        class="link btn btn--naked btn--size-sm link--primary"
        data-testid="button"
        href="https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-v14.pdf"
        target="_blank"
        type="button"
      >
         
        <span
          class="btn__content"
        >
          Download
        </span>
      </a>
    </object>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Embed> should render component with basic props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <object
        class="embed embed--default"
        data="https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-v14.pdf"
        data-testid="embed"
        type="application/pdf"
      >
        <p
          class="text text--size-md"
          data-testid="text"
        >
          Se você não consegue visualizar este documento, clique no botão abaixo e faça download!
        </p>
        <a
          class="link btn btn--primary btn--size-sm link--primary"
          data-testid="button"
          href="https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-v14.pdf"
          target="_blank"
          type="button"
        >
           
          <span
            class="btn__content"
          >
            Baixar arquivo
          </span>
        </a>
      </object>
    </div>
  </body>,
  "container": <div>
    <object
      class="embed embed--default"
      data="https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-v14.pdf"
      data-testid="embed"
      type="application/pdf"
    >
      <p
        class="text text--size-md"
        data-testid="text"
      >
        Se você não consegue visualizar este documento, clique no botão abaixo e faça download!
      </p>
      <a
        class="link btn btn--primary btn--size-sm link--primary"
        data-testid="button"
        href="https://cdn.mesalva.com/uploads/regulamentos/regulamento-resgate-de-livros-e-apostilas-v14.pdf"
        target="_blank"
        type="button"
      >
         
        <span
          class="btn__content"
        >
          Baixar arquivo
        </span>
      </a>
    </object>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
